kernel/kheap.c

```diff
@@ -116,6 +116,60 @@ void *k_heap_alloc(struct k_heap *heap, size_t bytes, k_timeout_t timeout)
 	return ret;
 }
 
+void *k_heap_aligned_realloc(struct k_heap *heap, void *ptr, size_t align, size_t bytes,
+			k_timeout_t timeout)
+{
+	k_timepoint_t end = sys_timepoint_calc(timeout);
+	void *ret = NULL;
+
+	k_spinlock_key_t key = k_spin_lock(&heap->lock);
+
+	SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_heap, aligned_alloc, heap, timeout);
+
+	__ASSERT(!arch_is_in_isr() || K_TIMEOUT_EQ(timeout, K_NO_WAIT), "");
+
+	bool blocked_realloc = false;
+
+	while (ret == NULL) {
+		ret = sys_heap_aligned_realloc(&heap->heap, ptr, align, bytes);
+
+		if (!IS_ENABLED(CONFIG_MULTITHREADING) ||
+		    (ret != NULL) || K_TIMEOUT_EQ(timeout, K_NO_WAIT)) {
+			break;
+		}
+
+		if (!blocked_realloc) {
+			blocked_realloc = true;
+
+			SYS_PORT_TRACING_OBJ_FUNC_BLOCKING(k_heap, aligned_realloc, heap, timeout);
+		} else {
+			/**
+			 * @todo	Trace attempt to avoid empty trace segments
+			 */
```

Comment:
I guess this is just a cut/paste of the same code in k_heap_aligned_alloc.  That said, this seems silly.   Just remove the else and add a single line /* TODO ... */ if really required.  But I guess I don't see value of tracing something you **didn't** do?

No complaints if you want to leave it as it is, but maybe remove the other one for symmetry if you agree.

Reply:
An extended version of that comment is that this is basically an identical function to k_heap_aligned_alloc() modulo what function to call in the middle, and it would be nice to seem them unified somehow via creating inlining or preprocessory.  But not required.